dotnet/src/Connectors/Connectors.OpenAI/Configuration/AzureOpenAIConfig.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using Azure.Core;
+
+namespace Microsoft.SemanticKernel;
+
+public class AzureOpenAIConfig
+{
+    private TokenCredential? _tokenCredential;
+
+    public enum AuthTypes
+    {
+        Unknown = -1,
+        AzureIdentity,
+        APIKey,
+        ManualTokenCredential,
+    }
+
+    public enum APITypes
+    {
+        Unknown = -1,
+        TextCompletion,
+        ChatCompletion,
+        ImageGeneration,
+        EmbeddingGeneration,
+    }
+
+    /// <summary>
+    /// OpenAI API type, e.g. text completion, chat completion, image generation, etc.
+    /// </summary>
+    public APITypes APIType { get; set; } = APITypes.ChatCompletion;
+
+    /// <summary>
+    /// Azure authentication type.
+    /// </summary>
+    public AuthTypes Auth { get; set; }
```

Comment:
As already mentioned, it would be nice to avoid using these enums. For `AuthTypes` - define specific classes. For `APITypes` - I'm not sure we need it at all.

Reply:
Please see my [comment](https://github.com/microsoft/semantic-kernel/pull/6283#issuecomment-2117053217) below.

Reply:
As requested I created specific classes for the configuration and removed the enums. Please take a look.